### DISCUSSION TITLE
How to run a Java program

### DISCUSSION THREAD

**Discussion by LittleMonster-LHQ** [thebaselab/codeapp#418] [General]
### I run the Java file when the console prompts codeapp running t.java remotely connection failed

**Comment by bummoblizard** [comment]
Are you running the TestFlight version? TestFlight builds don't support remote execution (to make it fair for paid users).

**Reply by LittleMonster-LHQ** [reply]
Thanks, I would like to know how to link remotely，I don't have a username and password for ftp

**Reply by bummoblizard** [reply]
You need to setup a ftp server first. You can also use sftp if your computer supports ssh.

Guide on enabling ssh access on a Mac: https://support.apple.com/en-gb/guide/mac-help/mchlp1066/mac

**Reply by LittleMonster-LHQ** [reply]
Thank you!!
